clients/client-sdk/src/clients/identity.ts

```diff
@@ -111,6 +130,85 @@ export class IdentityClient extends BaseClient {
     });
   }
 
+  private async restoreIdentity(
+    identity: IdentityKeys
+  ): Promise<{ document: Document; key: KeyPair; messageId: string }> {
+    try {
+      const decodedKey = {
+        public: Array.from(bs58.decode(identity.keys.sign.public)),
+        secret: Array.from(bs58.decode(identity.keys.sign.private))
+      };
+      const json = {
+        type: identity.keys.sign.type,
+        public: decodedKey.public,
+        private: decodedKey.secret
+      };
+      const key: KeyPair = KeyPair.fromJSON(json);
+      const { document, messageId } = await this.latestDocument(identity.id);
+      return {
+        document: Document.fromJSON(document),
+        key,
+        messageId
+      };
+    } catch (error) {
+      console.error(`Error from identity sdk: ${error}`);
+      throw new Error('could not parse key or doc of the identity');
+    }
+  }
+  /**
+   * Create a Verifiable Presentation.
+   *
+   * @param {{
+   *     signedVcJson: any;
+   *     identityKeys: IdentityKeys;
+   *     challenge?: string;
+   *     expiration?: number;
+   *   }} props Properties
+   * @param signedVcJson: Signed Verifiable Credential.
+   * @param identityKeys: Identity keys to sign the Verifiable Presentation.
+   * @param challenge: Challenge to mitigate replay attacks.
+   * @param expiration: Time when the presentation shall expire in seconds.
+   * @return {*}  {Promise<void>}
+   * @memberof IdentityClient
+   */
+  async createVerifiablePresentation(props: {
+    signedVcJson: any | any[];
+    identityKeys: IdentityKeys;
+    challenge?: string;
+    expiration?: number;
+  }): Promise<void> {
+    const { signedVcJson, identityKeys, challenge, expiration } = props;
+    const expires =
+      expiration != null ? Timestamp.nowUTC().checkedAdd(Duration.seconds(expiration)) : undefined;
+    const identity = await this.restoreIdentity(identityKeys);
+
+    let receivedVC: Credential | Credential[];
+    if (Array.isArray(signedVcJson)) {
+      receivedVC = signedVcJson.map((vc) => Credential.fromJSON(vc));
+    } else {
+      receivedVC = Credential.fromJSON(signedVcJson);
+    }
+
+    const pres: IPresentation = {
+      verifiableCredential: receivedVC,
+      holder: identity.document.id()
+    };
+
+    const unsignedVp = new Presentation(pres);
+    const methodId = identity.document.defaultSigningMethod().id().toString();
```

Comment:
I thought all requests to the `identity-wasm` package should be in the APIs. Are we going to have everything in the client that doesn't use the db?

Reply:
we cant do this on api side since it requires the private signature key of the identity, that's why we moved it into the frontend..